server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java

```diff
@@ -413,9 +413,10 @@ public CompletableFuture<Revision> updatePerRolePermissions(Author author,
 
         if (Project.REPO_META.equals(repoName)) {
             final Set<Permission> guest = perRolePermissions.guest();
-            if (!guest.isEmpty()) {
+            final Set<Permission> anonymous = perRolePermissions.anonymous();
+            if (!guest.isEmpty() || !anonymous.isEmpty()) {
                 throw new UnsupportedOperationException(
-                        "Can't give a permission to guest for internal repository: " + repoName);
+                        "Can't give a permission to guest or anonymous for internal repository: " + repoName);
             }
         }
 
```

Comment:
Could you also check if the anonymous permission has the write permission? If so, we should raise an exception.

Reply:
I've fixed to throw exception when trying to grant WRITE permission to anonymous user